developer/src/kmcmplib/tests/kmcompxtest.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * Tiny frame console app to compile a .kmx from a .kmn
+ */
+
+#include <windows.h>
+#include <stdio.h>
+#include "kmcompx.h"
+#include <iostream>
+#include <vector>
+#include <string>
+#include <sstream>
+using namespace std;
+
+vector < int > error_vec;
+
+int WINAPI msgproc(int line, KMX_DWORD dwMsgCode, char* szText)
+{
+  error_vec.push_back(dwMsgCode);
+  printf("line %d  error %x  %s\n", line, (unsigned int)dwMsgCode, szText);
+	return 1;
+}
+
+int main(int argc, char *argv[])
+{
+  PKMX_STR kmn_file = argv[1];
+  PKMX_STR kmx_file = argv[2];
+
+	if(argc < 3) {
+		puts("Usage: kmcompxtest source.kmn compiled.kmx");
+		return 1;
+	}
+
+	for(char *p = kmn_file; *p; p++) {
+		if(*p == '/') *p = '\\';
+	}
+
+	for(char *p = kmx_file; *p; p++) {
+		if(*p == '/') *p = '\\';
+	}
+
+	puts(kmn_file);
+	puts(kmx_file);
+
+  char  first5[6] = "CERR_";
+  char* pfirst5 = first5;
+
+  if (kmcmp_CompileKeyboardFile(kmn_file, kmx_file, FALSE, FALSE, TRUE, msgproc)) {
+    char* testname = strrchr( (char*) kmn_file, '\\') + 1;
+    if (strncmp(testname, pfirst5, 5) == 0) return 1; //no Error found + CERR_ in Name
+
+    // TODO: compare kmx_file to ../build/kmx_file
+    FILE* fp1 = fopen(kmx_file, "rb");
+    char fname[260];
+    strcpy(fname, kmx_file);
+    kmx_file[strlen(kmx_file)-1] = 'x'; // .kmn->.kmx
```

Comment:
I don't understand this code. Isn't the last character of `kmx_file` already `x`?  And why do we change it if we don't do anything with it?

Reply:
restored to older version; compares build/*.kmx to source/*.kmx agin; runs OK now